packages/stake/src/stake.ts

```diff
@@ -45,6 +46,10 @@ export const getUserJuiceDelta = async (
   to?: string | number,
   options?: Options,
 ): Promise<BigNumber> => {
+  const parsedFrom = await parseBlockTagToBlockNumber(from || epoch, options)
+  const parsedTo = await parseBlockTagToBlockNumber(to || 'latest', options)
+  const blockThreshold = 2000
```

Comment:
Well 2k is maximum limit, basic and even our plan has 1k limit whenever that kicks in. So default should be 1000 and that too configurable via options.

Reply:
Configurable option such that it may even be possible for client to try the old way first (blockTheshold = Infinity) and if that fails, go for split requests.